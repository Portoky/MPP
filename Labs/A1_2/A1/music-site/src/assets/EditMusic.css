.input {
  margin-left: 15px;
}
button {
  margin-left: 15px;
}
