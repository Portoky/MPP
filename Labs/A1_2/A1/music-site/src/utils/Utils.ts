/*import { Music } from "../entities/Music";

let guid = 0;
const getAndIncrementGuid = function (): number {
  guid = guid + 1;
  return guid - 1;
};

export const createMusic = (
  title: string,
  artist: string,
  rating: number,
  yearOfRelease: number
): Music => {
  const serialId = getAndIncrementGuid();
  return { serialId, title, artist, rating, yearOfRelease };
};
*/
export const elementsByPage = 5;
