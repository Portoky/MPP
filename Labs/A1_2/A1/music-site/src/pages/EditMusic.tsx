import { ChangeEvent } from "react";
import { Music } from "../entities/Music";
import Rating from "@mui/material/Rating";
import { useNavigate, useParams } from "react-router-dom";
import { useState } from "react";
import { MusicContext } from "../context/MusicContext";
import { useContext } from "react";
import "../assets/EditMusic.css";

const EditMusic = () => {
  const { musics, setMusics } = useContext(MusicContext);

  const navigate = useNavigate();

  const param = useParams();
  const stringSerialId = param["id"] || "-1";
  const serialId = parseInt(stringSerialId);

  const newMusics = musics.slice();
  const musicIndex = newMusics.findIndex((music: Music) => {
    return music.serialId === serialId;
  });
  if (serialId === -1) {
    return <p>Element not found!</p>;
  }

  const [artist, setartist] = useState(newMusics[musicIndex].artist);
  const [title, setTitle] = useState(newMusics[musicIndex].title);
  const [rating, setRating] = useState(newMusics[musicIndex].rating);
  const [yearOfRelease, setYearOfRelease] = useState(
    newMusics[musicIndex].yearOfRelease
  );

  const handleArtist = (event: ChangeEvent<HTMLInputElement>) => {
    setartist(event.target.value);
  };

  const handleTitle = (event: ChangeEvent<HTMLInputElement>) => {
    setTitle(event.target.value);
  };

  const handleYearOfRelease = (event: ChangeEvent<HTMLInputElement>) => {
    const inputNumber = parseInt(event.target.value);
    setYearOfRelease(isNaN(inputNumber) ? -1 : inputNumber);
  };

  const handleEditButtonClick = async () => {
    if (
      title == "" ||
      artist == "" ||
      yearOfRelease > 2024 ||
      yearOfRelease < 1000
    ) {
      alert("Invalid Music!");
      return;
    }

    const postData = {
      title: title,
      artist: artist,
      rating: rating,
      yearOfRelease: yearOfRelease,
    };
    await fetch("http://localhost:8080/edit/" + stringSerialId, {
      method: "PUT",
      body: JSON.stringify(postData),
      headers: {
        "Content-type": "application/json; charset=UTF-8",
      },
    })
      .then((response) => response.json())
      .then((data) => {
        console.log(data);
        navigate("/");
      })
      .catch((err) => {
        alert(err.message);
      });

    //newMusics[musicIndex].artist = artist;
    //newMusics[musicIndex].title = title;
    //newMusics[musicIndex].rating = rating;
    //newMusics[musicIndex].yearOfRelease = yearOfRelease;
    //setMusics(newMusics);
  };

  return (
    <>
      <h2>Edit music information with serialNumber -&gt; {serialId}</h2>
      <div className="input">
        <div className="input-item">
          <label>Artist: </label> <br></br>
          <input
            type="text"
            id="artist"
            name="artist"
            defaultValue={newMusics[musicIndex].artist}
            onChange={handleArtist}
          ></input>
        </div>
        <br></br>
        <div className="input-item">
          <label>Title: </label> <br></br>
          <input
            type="text"
            id="title"
            name="title"
            defaultValue={newMusics[musicIndex].title}
            onChange={handleTitle}
          ></input>
        </div>
        <br></br>
        <div className="input-item">
          <label>Rating: </label> <br></br>
          <Rating
            name="rating"
            value={rating}
            onChange={(event, newRating) => {
              setRating(newRating || 1);
            }}
          />
        </div>
        <br></br>
        <div className="input-item">
          <label>Release Year: </label> <br></br>
          <input
            type="number"
            id="yearOfRelease"
            name="yearOfRelease"
            defaultValue={newMusics[musicIndex].yearOfRelease}
            onChange={handleYearOfRelease}
          ></input>
        </div>
      </div>
      <br></br>
      <button
        type="button"
        className="btn btn-outline-success"
        onClick={handleEditButtonClick}
      >
        Edit Music
      </button>
    </>
  );
};

export default EditMusic;
