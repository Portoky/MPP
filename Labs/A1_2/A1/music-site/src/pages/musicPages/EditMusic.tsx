import { ChangeEvent } from "react";
import { Music } from "../../entities/Music";
import Rating from "@mui/material/Rating";
import { useNavigate, useParams } from "react-router-dom";
import { useState, useContext } from "react";
import { MusicContext } from "../../context/MusicContext";
import { ArtistContext } from "../../context/ArtistContext";
import "../../assets/EditMusic.css";
import Select from "react-select";
import { ConnectionContext } from "../../context/ConnectionContext";
import { db } from "../../db/db";

const EditMusic = () => {
  const { musics } = useContext(MusicContext);
  const { artists } = useContext(ArtistContext);
  const { isConnection } = useContext(ConnectionContext);
  const navigate = useNavigate();

  const param = useParams();
  const stringmusicId = param["id"] || "-1";
  const musicId = parseInt(stringmusicId);

  const newMusics = musics.slice();
  const musicIndex = newMusics.findIndex((music: Music) => {
    return music.musicId === musicId;
  });
  if (musicId === -1) {
    return <p>Element not found!</p>;
  }

  // eslint-disable-next-line react-hooks/rules-of-hooks
  const [artistId, setArtistId] = useState(-1);
  // eslint-disable-next-line react-hooks/rules-of-hooks
  const [title, setTitle] = useState(newMusics[musicIndex].title);
  // eslint-disable-next-line react-hooks/rules-of-hooks
  const [rating, setRating] = useState(newMusics[musicIndex].rating);
  // eslint-disable-next-line react-hooks/rules-of-hooks
  const [yearOfRelease, setYearOfRelease] = useState(
    newMusics[musicIndex].yearOfRelease
  );

  const handleArtist = (opt) => {
    const artistId = opt.value;
    setArtistId(artistId);
  };

  const handleTitle = (event: ChangeEvent<HTMLInputElement>) => {
    setTitle(event.target.value);
  };

  const handleYearOfRelease = (event: ChangeEvent<HTMLInputElement>) => {
    const inputNumber = parseInt(event.target.value);
    setYearOfRelease(isNaN(inputNumber) ? -1 : inputNumber);
  };

  const updateLocalDb = async () => {
    const postData = {
      title,
      artistId,
      rating,
      yearOfRelease,
    };
    try {
      db.musics.update(musicId, postData);
      navigate("/");
    } catch (error) {
      console.log("Couldnt update artist in local repo");
    }
  };

  const updateServerDb = async () => {
    const postData = {
      title: title,
      rating: rating,
      yearOfRelease: yearOfRelease,
    };
    await fetch(
      "http://localhost:8080/music/edit/" +
        stringmusicId +
        "/artist/" +
        artistId,
      {
        method: "PUT",
        body: JSON.stringify(postData),
        headers: {
          "Content-type": "application/json; charset=UTF-8",
        },
      }
    )
      .then((response) => response.json())
      .then(() => {
        navigate("/");
      })
      .catch((err) => {
        alert(err.message);
      });
  };

  const handleEditButtonClick = async () => {
    if (
      title === "" ||
      artistId === -1 ||
      yearOfRelease > 2024 ||
      yearOfRelease < 1000
    ) {
      alert("Invalid Music!");
      return;
    }
    if (isConnection === false) {
      updateLocalDb();
      return;
    }
    updateServerDb();
  };

  const artistOptions: { label: string; value: number }[] = [];
  artists.forEach((artist) => {
    artistOptions.push({ label: artist.name, value: artist.artistId });
  });
  return (
    <>
      <h2>Edit music information with serialNumber -&gt; {musicId}</h2>
      <div className="input">
        <div className="input-item">
          <label>Artist: </label> <br></br>
          <Select
            className="artistSelect"
            options={artistOptions}
            onChange={handleArtist}
          ></Select>
        </div>
        <br></br>
        <div className="input-item">
          <label>Title: </label> <br></br>
          <input
            type="text"
            className="form-control"
            id="title"
            name="title"
            defaultValue={newMusics[musicIndex].title}
            onChange={handleTitle}
          ></input>
        </div>
        <br></br>
        <div className="input-item">
          <label>Rating: </label> <br></br>
          <Rating
            name="rating"
            value={rating}
            onChange={(_event, newRating) => {
              setRating(newRating || 1);
            }}
          />
        </div>
        <br></br>
        <div className="input-item">
          <label>Release Year: </label> <br></br>
          <input
            type="number"
            className="form-control"
            id="yearOfRelease"
            name="yearOfRelease"
            defaultValue={newMusics[musicIndex].yearOfRelease}
            onChange={handleYearOfRelease}
          ></input>
        </div>
      </div>
      <br></br>
      <button
        type="button"
        style={{ marginLeft: "10px" }}
        className="btn btn-outline-success"
        onClick={handleEditButtonClick}
      >
        Edit Music
      </button>
    </>
  );
};

export default EditMusic;
