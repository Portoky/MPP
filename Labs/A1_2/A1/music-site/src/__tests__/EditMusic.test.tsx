import { describe, expect, test } from "vitest";
import { render } from "@testing-library/react";
import EditMusic from "../pages/EditMusic";
import { Routes, Route } from "react-router-dom";
import { createMusic } from "../utils/Utils";
import { MemoryRouter } from "react-router-dom";
describe("Edit", () => {
  test("renders", () => {
    const rendering = render(
      <MemoryRouter initialEntries={["/edit/1"]}>
        <Routes>
          <Route
            path="/edit/:id"
            element={
              <EditMusic
                musics={[
                  createMusic("Rocky Racoon", "The Beatles", 5, 1966),
                  createMusic("Nowhere Man", "The Beatles", 4, 1965),
                  createMusic("Let It Be", "The Beatles", 4, 1970),
                ]}
                setMusics={() => {}}
              ></EditMusic>
            }
          />
        </Routes>
      </MemoryRouter>
    );
    expect(rendering).toMatchSnapshot();
    //expect(screen.getByText("Music forum")).toBeDefined();
  });
});
