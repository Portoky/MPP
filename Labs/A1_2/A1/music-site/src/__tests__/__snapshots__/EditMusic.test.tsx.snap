// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Edit > renders 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2>
        Edit music information with serialNumber -&gt; 
        1
      </h2>
      <div
        class="input"
        style="margin-left: 15px;"
      >
        <div
          class="input-item"
        >
          <label>
            Artist: 
          </label>
           
          <br />
          <input
            id="artist"
            name="artist"
            type="text"
            value="The Beatles"
          />
        </div>
        <br />
        <div
          class="input-item"
        >
          <label>
            Title: 
          </label>
           
          <br />
          <input
            id="title"
            name="title"
            type="text"
            value="Nowhere Man"
          />
        </div>
        <br />
        <div
          class="input-item"
        >
          <label>
            Rating: 
          </label>
           
          <br />
          <span
            class="MuiRating-root MuiRating-sizeMedium css-1qqgbpl-MuiRating-root"
          >
            <label
              class="css-dqr9h-MuiRating-label"
              for=":r1:"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                1 Star
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":r1:"
              name="rating"
              type="radio"
              value="1"
            />
            <label
              class="css-dqr9h-MuiRating-label"
              for=":r2:"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                2 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":r2:"
              name="rating"
              type="radio"
              value="2"
            />
            <label
              class="css-dqr9h-MuiRating-label"
              for=":r3:"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                3 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":r3:"
              name="rating"
              type="radio"
              value="3"
            />
            <label
              class="css-dqr9h-MuiRating-label"
              for=":r4:"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                4 Stars
              </span>
            </label>
            <input
              checked=""
              class="MuiRating-visuallyHidden"
              id=":r4:"
              name="rating"
              type="radio"
              value="4"
            />
            <label
              class="css-dqr9h-MuiRating-label"
              for=":r5:"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyHidden"
              >
                5 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyHidden"
              id=":r5:"
              name="rating"
              type="radio"
              value="5"
            />
            <label
              class="MuiRating-label MuiRating-pristine css-dqr9h-MuiRating-label"
            >
              <input
                class="MuiRating-visuallyHidden"
                id="rating-empty"
                name="rating"
                type="radio"
                value=""
              />
              <span
                class="MuiRating-visuallyHidden"
              >
                Empty
              </span>
            </label>
          </span>
        </div>
        <br />
        <div
          class="input-item"
        >
          <label>
            Release Year: 
          </label>
           
          <br />
          <input
            id="yearOfRelease"
            name="yearOfRelease"
            type="number"
            value="1965"
          />
        </div>
      </div>
      <br />
      <button
        class="btn btn-outline-success"
        style="margin-left: 15px;"
        type="button"
      >
        Edit Music
      </button>
    </div>
  </body>,
  "container": <div>
    <h2>
      Edit music information with serialNumber -&gt; 
      1
    </h2>
    <div
      class="input"
      style="margin-left: 15px;"
    >
      <div
        class="input-item"
      >
        <label>
          Artist: 
        </label>
         
        <br />
        <input
          id="artist"
          name="artist"
          type="text"
          value="The Beatles"
        />
      </div>
      <br />
      <div
        class="input-item"
      >
        <label>
          Title: 
        </label>
         
        <br />
        <input
          id="title"
          name="title"
          type="text"
          value="Nowhere Man"
        />
      </div>
      <br />
      <div
        class="input-item"
      >
        <label>
          Rating: 
        </label>
         
        <br />
        <span
          class="MuiRating-root MuiRating-sizeMedium css-1qqgbpl-MuiRating-root"
        >
          <label
            class="css-dqr9h-MuiRating-label"
            for=":r1:"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
            <span
              class="MuiRating-visuallyHidden"
            >
              1 Star
            </span>
          </label>
          <input
            class="MuiRating-visuallyHidden"
            id=":r1:"
            name="rating"
            type="radio"
            value="1"
          />
          <label
            class="css-dqr9h-MuiRating-label"
            for=":r2:"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
            <span
              class="MuiRating-visuallyHidden"
            >
              2 Stars
            </span>
          </label>
          <input
            class="MuiRating-visuallyHidden"
            id=":r2:"
            name="rating"
            type="radio"
            value="2"
          />
          <label
            class="css-dqr9h-MuiRating-label"
            for=":r3:"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
            <span
              class="MuiRating-visuallyHidden"
            >
              3 Stars
            </span>
          </label>
          <input
            class="MuiRating-visuallyHidden"
            id=":r3:"
            name="rating"
            type="radio"
            value="3"
          />
          <label
            class="css-dqr9h-MuiRating-label"
            for=":r4:"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
            <span
              class="MuiRating-visuallyHidden"
            >
              4 Stars
            </span>
          </label>
          <input
            checked=""
            class="MuiRating-visuallyHidden"
            id=":r4:"
            name="rating"
            type="radio"
            value="4"
          />
          <label
            class="css-dqr9h-MuiRating-label"
            for=":r5:"
          >
            <span
              class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarBorderIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                />
              </svg>
            </span>
            <span
              class="MuiRating-visuallyHidden"
            >
              5 Stars
            </span>
          </label>
          <input
            class="MuiRating-visuallyHidden"
            id=":r5:"
            name="rating"
            type="radio"
            value="5"
          />
          <label
            class="MuiRating-label MuiRating-pristine css-dqr9h-MuiRating-label"
          >
            <input
              class="MuiRating-visuallyHidden"
              id="rating-empty"
              name="rating"
              type="radio"
              value=""
            />
            <span
              class="MuiRating-visuallyHidden"
            >
              Empty
            </span>
          </label>
        </span>
      </div>
      <br />
      <div
        class="input-item"
      >
        <label>
          Release Year: 
        </label>
         
        <br />
        <input
          id="yearOfRelease"
          name="yearOfRelease"
          type="number"
          value="1965"
        />
      </div>
    </div>
    <br />
    <button
      class="btn btn-outline-success"
      style="margin-left: 15px;"
      type="button"
    >
      Edit Music
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
